backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardlist/BpkCardList.kt

```diff
@@ -0,0 +1,57 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.cardlist
+
+import androidx.compose.foundation.lazy.LazyItemScope
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import net.skyscanner.backpack.compose.cardlist.internal.BpkCardListImpl
+
+@Composable
+fun <T> BpkCardList(
+    title: String,
+    description: String,
+    layout: BpkCardListLayout,
+    dataList: List<T>,
+    modifier: Modifier = Modifier,
+    elements: @Composable (LazyItemScope.(Int) -> Unit),
+) {
+    BpkCardListImpl(
+        title = title,
+        description = description,
+        layout = layout,
+        modifier = modifier,
+        dataList = dataList,
+        elements = elements,
+    )
+}
+
+sealed class BpkCardListLayout(
+    open val button: BpkCardListButtonAccessory?,
```

Comment:
looking at the api in figma this is not correct. there should be a button prop on the top level and in the stack case there's either a button or expand option.
as you're not implementing the stack option right now i would just leave this out and if we add the stack option later on the default will just be rail to have a non-breaking change

Reply:
I got your point but I think we will still have a breaking change for stack later.

from the API definition, we can not have a header button and an accessory button at the same time. If we keep the API definitions we could have both being instantiated but which one would have priority to be rendered?

Reply:
I think you'd show both in the stack case: the accessory button is for the end of the stack - not the button for the section header

Reply:
![Screenshot 2024-05-24 at 17 08 18](https://github.com/Skyscanner/backpack-android/assets/12631700/6eb4074e-0ff1-4029-ab10-6b6da83ef6e5)
thats the issue 😄

Reply:
Ah I see. In that case we just don't show the accessory button even if you set the accessory type to button. I don't think we can really control this from an api perspective without being too complex as there's a bunch of combinations. For example you might ONLY want an accessory button and not a section header button - having the button in the layout property would then not handle that.
I'm also not 100% sure if the button accessory option is even a case on apps as I can't see any designs for it (just the desktop version).
So to simplify the API & usage i would keep the section header button in the top level as that's clearer where it'll be shown.

Reply:
cool, I'll raise a thread in the group later - because we will need to align with iOS too 😄